test/remote/gateways/remote_shift4_v2_test.rb

```diff
@@ -0,0 +1,341 @@
+require 'test_helper'
+
+class RemoteShift4V2Test < Test::Unit::TestCase
+  CHARGE_ID_REGEX = /char_[a-zA-Z\d]+/
+  TOKEN_ID_REGEX  = /tok_[a-zA-Z\d]+/
+
+  def setup
+    @gateway = Shift4V2Gateway.new(fixtures(:shift4_v2))
+
+    @amount = 2000
+    @refund_amount = 300
+    @credit_card = credit_card('4242424242424242')
+    @declined_card = credit_card('4916018475814056')
+    @new_credit_card = credit_card('4012888888881881')
+    @invalid_token = 'tok_invalid'
+
+    @options = {
+      description: 'ActiveMerchant test charge',
+      email: 'foo@example.com'
+    }
+  end
+
+  def test_successful_store
+    response = @gateway.store(@credit_card, @options)
+    assert_success response
+    assert_match %r(^cust_\w+$), response.authorization
+    assert_equal 'customer', response.params['objectType']
+    assert_match %r(^card_\w+$), response.params['cards'][0]['id']
+    assert_equal 'card', response.params['cards'][0]['objectType']
+
+    @options[:customer_id] = response.authorization
+    response = @gateway.store(@new_credit_card, @options)
+    assert_success response
+    assert_match %r(^card_\w+$), response.params['card']['id']
+    assert_equal @options[:customer_id], response.params['card']['customerId']
+
+    response = @gateway.customer(@options)
+    assert_success response
+    assert_equal @options[:customer_id], response.params['id']
+    assert_equal '401288', response.params['cards'][0]['first6']
+    assert_equal '1881', response.params['cards'][0]['last4']
+    assert_equal '424242', response.params['cards'][1]['first6']
+    assert_equal '4242', response.params['cards'][1]['last4']
+  end
+
+  # def test_dump_transcript
+  #   skip("Transcript scrubbing for this gateway has been tested.")
+  #   dump_transcript_and_fail(@gateway, @amount, @credit_card, @options)
+  # end
```

Comment:
Will this code be needed later? or can we remove it?

Reply:
Seems that is just trash